src/transformers/models/imagebind/modeling_imagebind.py

```diff
@@ -0,0 +1,2062 @@
+        return tensor.view(batch_size, seq_len, self.num_heads, self.head_dim).transpose(1, 2).contiguous()
+
+    def forward(
+        self,
+        hidden_states: torch.Tensor,
+        attention_mask: Optional[torch.Tensor] = None,
+        output_attentions: Optional[bool] = False,
+    ) -> Tuple[torch.Tensor, Optional[torch.Tensor], Optional[Tuple[torch.Tensor]]]:
+        """Input shape: Batch x Time x Channel"""
+
+        batch_size, seq_len, embed_dim = hidden_states.size()
+
+        qkv = self.qkv_proj(hidden_states).reshape(batch_size, seq_len, 3, -1).permute(2, 0, 1, 3)
+        query_states, key_states, value_states = qkv.unbind(0)
+
+        query_states = query_states * self.scale
+
+        # Add key/value biases if necessary
+        if self.add_kv_bias:
+            # Repeat bias along batch dimension (first)
+            key_states = torch.cat([key_states, self.k_bias.repeat(batch_size, 1, 1)], dim=1)
+            value_states = torch.cat([value_states, self.v_bias.repeat(batch_size, 1, 1)], dim=1)
+
+        key_states = self._shape(key_states, -1, batch_size)
+        value_states = self._shape(value_states, -1, batch_size)
+
+        proj_shape = (batch_size * self.num_heads, -1, self.head_dim)
+        query_states = self._shape(query_states, seq_len, batch_size).view(*proj_shape)
+        key_states = key_states.view(*proj_shape)
+        value_states = value_states.view(*proj_shape)
+
+        src_len = key_states.size(1)
+        attn_weights = torch.bmm(query_states, key_states.transpose(1, 2))
+
+        if attn_weights.size() != (batch_size * self.num_heads, seq_len, src_len):
+            raise ValueError(
+                f"Attention weights should be of size {(batch_size * self.num_heads, seq_len, src_len)}, but is"
+                f" {attn_weights.size()}"
+            )
+
+        if attention_mask is not None:
+            if attention_mask.size() != (batch_size, 1, seq_len, src_len):
+                raise ValueError(
+                    f"Attention mask should be of size {(batch_size, 1, seq_len, src_len)}, but is {attention_mask.size()}"
+                )
+            attn_weights = attn_weights.view(batch_size, self.num_heads, seq_len, src_len) + attention_mask
+            attn_weights = attn_weights.view(batch_size * self.num_heads, seq_len, src_len)
+
+        attn_weights = nn.functional.softmax(attn_weights, dim=-1)
+
+        if output_attentions:
+            # this operation is a bit akward, but it's required to
+            # make sure that attn_weights keeps its gradient.
+            # In order to do so, attn_weights have to reshaped
+            # twice and have to be reused in the following
+            attn_weights_reshaped = attn_weights.view(batch_size, self.num_heads, seq_len, src_len)
+            attn_weights = attn_weights_reshaped.view(batch_size * self.num_heads, seq_len, src_len)
+        else:
+            attn_weights_reshaped = None
```

Comment:
yeah this is no longer needed and when I digged a bit it was du to naming issues

Reply:
@ Is the issue been fixed on main? If it is then I will update with the same thing here as well for consistency

Reply:
Yep it was fixed see UMT5: 
```python
        # (batch_size, n_heads, seq_length, key_length)
        attn_weights = nn.functional.softmax(scores.float(), dim=-1).type_as(scores)
        attn_weights = nn.functional.dropout(attn_weights, p=self.dropout, training=self.training)

        # Mask heads if we want to
        if layer_head_mask is not None:
            attn_weights = attn_weights * layer_head_mask

        attn_output = torch.matmul(attn_weights, value_states)

        attn_output = attn_output.transpose(1, 2).contiguous()
        attn_output = attn_output.view(batch_size, seq_length, -1)

        attn_output = self.o(attn_output)
        return attn_output, attn_weights, past_key_value
```
I based it of T5, did not propagate to T5 but I can it's a good idea! 😉